client/packages/system/src/Patient/CreatePatientModal/CreatePatientModal.tsx

```diff
@@ -56,7 +58,19 @@ export const CreatePatientModal: FC<CreatePatientModal> = ({ onClose }) => {
 
   const onOk = () => {
     if (createNewPatient) {
-      navigate(createNewPatient.id);
+      const urlSegments = location.pathname.split('/');
+
+      if (urlSegments.includes(AppRoute.Patients))
+        navigate(createNewPatient.id);
+
+      if (urlSegments.includes(AppRoute.Prescription))
+        navigate(
+          RouteBuilder.create(AppRoute.Dispensary)
+            .addPart(AppRoute.Patients)
+            .addPart(createNewPatient.id)
+            .addQuery({ previousPath: AppRoute.Prescription })
+            .build()
```

Comment:
Just wondering if the main part of the url should stay the same either way. We have the query `?previousPath=prescription` to determine where to head back to, so maybe simpler to keep the base url the same?

Reply:
And then you could just use the `updateQuery` method of `useUrlQuery` hook to add to the url and leave the main builder untouched.

Reply:
I've tried looking into the suggestion! Though I don't think we can keep the main part of the url to stay the same as the modal gets rendered in different pages. And have to redirect to patients page from prescriptions

Reply:
Chatted in person and have resolved